src/components/hooks/useCatalogsForSubsidyRequests.js

```diff
@@ -0,0 +1,47 @@
+import { useQuery } from '@tanstack/react-query';
+import {
+  queryBrowseAndRequestConfiguration,
+  queryCouponCodes,
+  querySubscriptions,
+} from '../app/data';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+
+import { SUBSIDY_TYPE } from '../../constants';
+
+export default function useCatalogForSubsidyRequest() {
+  const { uuid } = useEnterpriseCustomer();
+
+  const { data: browseAndRequestsData } = useQuery(queryBrowseAndRequestConfiguration(uuid));
+  const { data: couponCodesData } = useQuery(queryCouponCodes(uuid));
+  const { data: subscriptionsData } = useQuery(querySubscriptions(uuid));
+  const { subscriptionLicenses: subscriptionLicensesData } = subscriptionsData;
+  const { couponsOverview: { results: couponsOverviewData } } = couponCodesData;
+
+  const catalogsForSubsidyRequests = [];
+
+  if (!browseAndRequestsData.subsidyRequestsEnabled) {
+    return { catalogsForSubsidyRequests };
+  }
+
+  if (browseAndRequestsData.subsidyType === SUBSIDY_TYPE.LICENSE) {
+    const catalogsFromSubscriptions = subscriptionLicensesData
+      .filter(
+        subscriptionLicense => subscriptionLicense.subscriptionPlan.daysUntilExpirationIncludingRenewals > 0,
+      )
+      .map(subscriptionLicense => subscriptionLicense.subscriptionPlan.enterpriseCatalogUuid);
+
+    catalogsForSubsidyRequests.push(...new Set(catalogsFromSubscriptions));
+  }
+
+  if (browseAndRequestsData.subsidyType === SUBSIDY_TYPE.COUPON) {
+    const catalogsFromCoupons = couponsOverviewData
+      .filter(coupon => !!coupon.available)
+      .map(coupon => coupon.enterpriseCatalogUuid);
+
+    catalogsForSubsidyRequests.push(...new Set(catalogsFromCoupons));
+  }
+
+  return {
+    catalogsForSubsidyRequests,
```

Comment:
Fine as is for now, but we _may_ end up needing to memoize the return to ensure the returned data does not have a new array reference on re-renders. Can be deferred until we run into a specific problem with it.

Reply:
FWIW its one of the possible issues I had in mind when refactoring this section of code is the need of memoizing some of the subsidy data. Decided against it in the end because we are depending on the query cache for these set of calls after the loader has brought all the data in with `ensureQueryData`.

Will leave open as a reminder if we do see issues in the future 👍🏽